### PR TITLE
Inherit table charset/collation from db charset

### PR DESCRIPTION
**Pull request by tvlooy**
If the database is utf8mb4, tables should be utf8mb4 by default.

### CODE REVIEW

**Comment by tvlooy** [comment]
@kimhemsoe I added a test

File: lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php [hunk 1]
```diff
@@ -1058,9 +1058,13 @@ public function createSchemaConfig()
         }
 
         $params = $this->_conn->getParams();
-        if (isset($params['defaultTableOptions'])) {
-            $schemaConfig->setDefaultTableOptions($params['defaultTableOptions']);
+        if (! isset($params['defaultTableOptions'])) {
+            $params['defaultTableOptions'] = [];
         }
+        if (! isset($params['defaultTableOptions']['charset']) && isset($params['charset'])) {
```
Comment (mikeSimonson) [inline lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php]: Should we test for emptiness ?

Comment (tvlooy) [reply lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php]: I guess if you set $params['charset'] to null, '', [], an object or ... you are deliberately trying to break stuff. Do I understand your question right?

**Comment by tvlooy** [comment]
Can we merge this?

**Comment by mikeSimonson** [comment]
@tvlooy Thanks

**Comment by tvlooy** [comment]
@mikeSimonson I don't know how doctrine's release process works but will this be in the next 2.6.* too? http://www.doctrine-project.org/contribute.html says that there are release-* branches but I don't see these in dbal. Is this the 2.6 branch?

**Comment by Majkl578** [comment]
New features are not backported to already stable releases, you will need to wait for 2.7.

**Comment by tvlooy** [comment]
but this is a bugfix

**Pull request merged**